dynamodb/src/main/scala/zio/dynamodb/Codec.scala

```diff
@@ -172,7 +172,10 @@ private[dynamodb] object Codec {
             val value              = valuesMap(key)
             val enc                = encoder[a](schema)
             val av: AttributeValue = enc(value.asInstanceOf[a])
-            AttributeValue.Map(avMap.value + (AttributeValue.String(key) -> av))
+            av match {
+              case AttributeValue.Null => avMap
+              case _                   => AttributeValue.Map(avMap.value + (AttributeValue.String(key) -> av))
+            }
```

Comment:
This codecs leads to generating "NULL: null" values in the result Dynamo document; which is not always parsable by other clients and also not aligned with the case classes behavior.